sdk/identity/src/token_credentials/client_secret_credentials.rs

```diff
@@ -4,9 +4,55 @@ use chrono::Utc;
 use oauth2::{
     basic::BasicClient, reqwest::async_http_client, AccessToken, AuthType, AuthUrl, Scope, TokenUrl,
 };
-use std::{str, time::Duration};
+use std::{borrow::Cow, str, time::Duration};
 use url::Url;
 
+/// Provides options to configure how the Identity library makes authentication
+/// requests to Azure Active Directory.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TokenCredentialOptions {
+    /// The authority host to use for authentication requests.  The default is
+    /// "https://login.microsoftonline.com".
+    pub authority_host: Option<Cow<'static, str>>,
+}
+
+const DEFAULT_TOKEN_CREDENTIAL_OPTIONS: TokenCredentialOptions = TokenCredentialOptions {
+    authority_host: Some(Cow::Borrowed(authority_hosts::AZURE_PUBLIC_CLOUD)),
+};
+
+impl Default for TokenCredentialOptions {
```

Comment:
Maybe we should add a `new` function since you have implemented `default` and then make the `authority_host` field non public.